lib/adapters/TapeAdapter.js

```diff
@@ -0,0 +1,134 @@
+const EventEmitter = require('events');
+
+module.exports = class TapeAdapter extends EventEmitter {
+  /**
+   * Unlike other adapters, this adapter ends up disabling the default
+   * reporter of its framework (in this case, Tape's TAP reporter to
+   * console.log). This is the default Tape behaviour, and we choose not to
+   * change that. If you wish to enable Tape's default reporter also, and
+   * e.g. use CRI only for additional reports to network or artefact files,
+   * then run `tape.createStream().pipe(process.stdout);` to have both.
+   *
+   * @param {Tape} tape
+   */
+  constructor (tape) {
+    super();
+
+    // TODO: Unable to observe 'prerun' event for some reason.
+    let started = false;
+    let runStartTime;
+    let runStatus = 'passed';
+    const runCounts = { passed: 0, failed: 0, skipped: 0, todo: 0, total: 0 };
+    const startOnce = () => {
+      if (!started) {
+        started = true;
+        this.emit('runStart', { name: null, counts: { total: null } });
+        runStartTime = new Date().getTime();
```

Comment:
```suggestion
        runStartTime = Date.now();
```

Reply:
I'm aiming to support IE9+ in the long-term here since imho this should be a simple low-level tool that won't stand in the way, especially e.g. with browserstack, saucelabs, and karma integration.

Having said that, this part of ES5 indeed landed pretty early and widely, so I guess we can just use that. I intuitively avoided it knowing it got spec'ed by ES5, and hadn't really considered when browsers implemented it. Thanks!

Reply:
I’m a huge fan of supporting old browsers, but old browsers will almost certainly be polyfilled.

Reply:
At what level though? At some point one can't "pass the buck" any further.

Reply:
i expect everything to have the equivalent of the es5-shim and es6-shim installed.